controllers/goroutines/operator_status.go

```diff
@@ -82,37 +83,64 @@ func UpdateCsCrStatus(bs *bootstrap.Bootstrap) {
 	}
 }
 
-func getBedrockOperator(bs *bootstrap.Bootstrap, name, namespace string) apiv3.BedrockOperator {
+func getBedrockOperator(bs *bootstrap.Bootstrap, name, namespace string) (apiv3.BedrockOperator, error) {
 	var opt apiv3.BedrockOperator
+	opt.Name = name
 
 	// fetch subscription
 	sub := &olmv1alpha1.Subscription{}
 	subKey := types.NamespacedName{
 		Name:      name,
 		Namespace: namespace,
 	}
-	if subErr := bs.Client.Get(ctx, subKey, sub); subErr != nil {
-		return opt
+	if err := bs.Client.Get(ctx, subKey, sub); err != nil {
+		return opt, err
 	}
-	installCSV := sub.Status.InstalledCSV
-	if installCSV != "" {
-		opt.Name = installCSV[:strings.IndexByte(installCSV, '.')]
-		opt.Version = installCSV[strings.IndexByte(installCSV, '.')+1:]
+	installedCSV := sub.Status.InstalledCSV
+	if installedCSV != "" {
+		opt.Name = installedCSV[:strings.IndexByte(installedCSV, '.')]
+		opt.Version = installedCSV[strings.IndexByte(installedCSV, '.')+1:]
 	}
 
 	// fetch csv
 	csv := &olmv1alpha1.ClusterServiceVersion{}
 	csvKey := types.NamespacedName{
-		Name:      installCSV,
+		Name:      installedCSV,
 		Namespace: namespace,
 	}
-	if csvErr := bs.Reader.Get(ctx, csvKey, csv); csvErr != nil {
-		return opt
+	if err := bs.Reader.Get(ctx, csvKey, csv); err != nil {
+		klog.Warningf("%s CSV %s", name, err)
+	}
+
+	// fetch installplanName
+	installplanName := ""
+	if sub.Status.Install != nil {
+		installplanName = sub.Status.Install.Name
 	}
+	opt.InstallPlanName = installplanName
+
+	// determinate subscription status
+	if installplanName == "" {
+		opt.SubscriptionStatus = "Failed"
+		opt.InstallPlanName = "Not Found"
+	} else {
+		currentCSV := sub.Status.CurrentCSV
+		installedIsLarger := util.CompareVersion(installedCSV, currentCSV)
+		currentIsLarger := util.CompareVersion(currentCSV, installedCSV)
```

Comment:
@ZhuoxiLi 

It is not accurate to compare the version of installedCSV and currentCSV to decide the operator status since numerically, operator can be upgraded to a CSV with a smaller number. OLM doesn't manage the upgrade by CSV number.

Could get the `opt.SubscriptionStatus` directly from subscription status. For example, if installedCSV equal to currentCSV then it is Succeeded, otherwise `opt.SubscriptionStatus=sub.status.state`

Reply:
@ZhuoxiLi 
`installedCSV` and `currentCSV` are strings.
Could we use `installedCSV == currentCSV` and bypass the `util.CompareVersion()`?

Reply:
ah, that's true, since we used need to know whether currentCSV > installedCSV, so need util.CompareVersion(). I will update it with compare the string.
other than that, I prefer to keep CompareVersion() in util, more flexible for future use.